x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -719,9 +737,12 @@ private synchronized boolean addSetStopAtCheckpointListener(
 
     synchronized void stopAndSaveState() {
```

Comment:
I know that this name has been the same for a while.

But I think something like `stopAndMaybeSaveState` is better is it will only save state if the indexer is NOT already shutting down + the indexer stopped.